src/xtgeo/grid3d/grid_property.py

```diff
@@ -1113,6 +1113,37 @@ def mask_undef(self) -> None:
         else:
             self._values = np.ma.masked_greater(self._values, UNDEF_LIMIT)
 
+    def refine(
+        self,
+        refine_col: int | list[int],
+        refine_row: int | list[int],
+        refine_layer: int | list[int],
+    ) -> None:
+        """
```

Comment:
I don't think that grid_property should have a public function `refine` here, at its would not work without the grid being refined also. You can make it a private method, possibly just in a grid_property sub module like `gridprop_op1.py`